### ISSUE TITLE
[2.x] Browser "Back" button does not return to the previous page with prefetch on mobile devices

### ISSUE THREAD

**Issue by ramonmalcolm10** [inertiajs/inertia#2122]
### Versions:

- `inertiajs/inertia-laravel` version: 2.x-dev
- `@inertiajs/react` version: 2.0.0-beta.3

### Describe the problem:

When using Inertia JS v2 on Chrome with an iPhone, I noticed an issue with navigation when using the prefetch feature. After navigating to a new page and clicking the browser "Back" button, it does not return to the previous page as expected. This behavior occurs consistently on this device and browser combination. I haven't tested this without prefetch or on other devices/browsers.

### Steps to reproduce:

1. Set up an Inertia.js v2 project.
2. Enable the prefetch feature for navigation links.
3. Open the app on an iPhone using Chrome.
4. Click on a link with prefetch enabled to navigate to a new page.
5. Press the browser "Back" button.
6. Observe the behavior.

**Comment by NicolasKion** [comment]
Looks like the fix from the 1.3 beta did not make it into the 2.x beta yet  https://github.com/inertiajs/inertia/releases/tag/v1.3.0-beta.1

**Comment by ramonmalcolm10** [comment]
Just a bit curious why was v2 marked as stable without  bug fixes from version 1.3 @NicolasKion

**Comment by HichemTab-tech** [comment]
> Just a bit curious why was v2 marked as stable without bug fixes from version 1.3 @NicolasKion

Yes @ramonmalcolm10 it is weird, also this [issue](https://github.com/inertiajs/inertia/issues/2068) is not fixed neither these PRs ([inertia-laravel](https://github.com/inertiajs/inertia-laravel/pull/679) & [inertia](https://github.com/inertiajs/inertia/pull/2069)), since it is a really important fix.

**Comment by rjsteiger** [comment]
Happening in Edge on iOS (iPhone/iPad) also. Prefetch was _not_ added as attribute to my inertia links `<Link prefetch></Link>`. Back button exits site completely and returns to browser home page, search engine, or page previously visited to my site. Same issue?

**Comment by ramonmalcolm10** [comment]
> Happening in Edge on iOS (iPhone/iPad) also. Prefetch was _not_ added as attribute to my inertia links `<Link prefetch></Link>`. Back button exits site completely and returns to browser home page, search engine, or page previously visited to my site. Same issue?
> 
> Works as expected in Safari and Firefox on iOS for me.

Seems it impact all browsers on iOS with the exception of Safari.

**Comment by rjsteiger** [comment]
I am seeing that this issue was fixed / closed and released on v2.02. I am running v2.06 and am still unable to navigate using the back button in Chrome/Edge on iOS. Does this continue happening to anyone else?

**Comment by ramonmalcolm10** [comment]
> I am seeing that this issue was fixed / closed and released on v2.02. I am running v2.06 and am still unable to navigate using the back button in Chrome/Edge on iOS. Does this continue happening to anyone else?

Yes I still experience the issue.

**Comment by rjsteiger** [comment]
> Yes I still experience the issue.

I wonder if this would be more of an inertia-laravel issue vs inertiajs/inertia. I am also using laravel, and I don't have the file packages/code/src/router.ts or packages/code/src/history.ts as were committed in the v2.02 release.

**Comment by moesuke-moekan** [comment]
It seems to be currently occurring with v2.0.13 as well.

**Comment by pascalbaljet** [comment]
I've fixed a race condition in the history manipulation logic that could cause issues with browser history state management. It's available starting in v2.2.7. Could you please update the Inertia library and confirm if the issue still exists or not?

**Comment by rjsteiger** [comment]
Upgraded to v2.2.7... Issue still persists.

**Comment by pascalbaljet** [comment]
I could reproduce it and it's fixed in v2.2.8.

**Comment by brandanking-decently** [comment]
I'm not to sure if this problem is directly related @pascalbaljet but when using the navigation buttons the page doesn't seem to actually change, my Inertia version is v2.2.9

https://github.com/user-attachments/assets/9a5624ed-659a-4e71-b98d-f3d3759848a8
